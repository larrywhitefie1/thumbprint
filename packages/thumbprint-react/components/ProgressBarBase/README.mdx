---
package: '@thumbtack/thumbprint-react'
kit: progress-bar-base/index.mdx
platform: react
url: /components/progress-bar-base/react/
mdxType: componentApi
---

```jsx
<ProgressBarBase width={70} />
```
