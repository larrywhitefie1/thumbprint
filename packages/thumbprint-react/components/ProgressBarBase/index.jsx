import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.module.scss';

const ProgressBarBase = ({ width }) => (
    <div className={styles.outer}>
        <div
            className={styles.inner}
            style={{ transform: `translateX(-${100 - width}%)` }}
            role="progressbar"
            aria-valuenow={width}
            aria-valuemin="0"
            aria-valuemax="100"
        />
    </div>
);

ProgressBarBase.propTypes = {
    /**
     * A number between 0 and 100.
     */
    width: PropTypes.number,
};

ProgressBarBase.defaultProps = {
    width: 0,
};

export default ProgressBarBase;
