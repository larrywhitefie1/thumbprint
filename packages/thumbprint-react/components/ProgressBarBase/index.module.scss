@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';

.outer {
    background-color: $tp-color__gray-200;
    border-radius: 3px;
    height: 6px;
    overflow: hidden;
}

.inner {
    background-color: $tp-color__blue;
    border-radius: 3px;
    height: 100%;
    width: 100%;
    transition: transform 1s ease;
}
