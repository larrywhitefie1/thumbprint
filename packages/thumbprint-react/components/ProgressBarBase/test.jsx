import React from 'react';
import { mount } from 'enzyme';
import ProgressBarBase from './index';

test('updates CSS transform based on `width`', () => {
    const wrapperA = mount(<ProgressBarBase width={33} />);
    expect(wrapperA).toMatchSnapshot();

    const wrapperB = mount(<ProgressBarBase width={50} />);
    expect(wrapperB).toMatchSnapshot();
});

test('updates `aria-valuenow` as `width` changes', () => {
    const wrapperA = mount(<ProgressBarBase width={33} />);
    const progressbarA = wrapperA.find('.inner');
    expect(progressbarA.prop('aria-valuenow')).toBe(33);
    expect(wrapperA).toMatchSnapshot();

    const wrapperB = mount(<ProgressBarBase width={50} />);
    const progressbarB = wrapperB.find('.inner');
    expect(progressbarB.prop('aria-valuenow')).toBe(50);
    expect(wrapperB).toMatchSnapshot();
});
