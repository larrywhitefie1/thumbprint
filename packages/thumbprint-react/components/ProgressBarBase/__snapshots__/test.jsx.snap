// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates \`aria-valuenow\` as \`width\` changes 1`] = `
<ProgressBarBase
  width={33}
>
  <div
    className="outer"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={33}
      className="inner"
      role="progressbar"
      style={
        Object {
          "transform": "translateX(-67%)",
        }
      }
    />
  </div>
</ProgressBarBase>
`;

exports[`updates \`aria-valuenow\` as \`width\` changes 2`] = `
<ProgressBarBase
  width={50}
>
  <div
    className="outer"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={50}
      className="inner"
      role="progressbar"
      style={
        Object {
          "transform": "translateX(-50%)",
        }
      }
    />
  </div>
</ProgressBarBase>
`;

exports[`updates CSS transform based on \`width\` 1`] = `
<ProgressBarBase
  width={33}
>
  <div
    className="outer"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={33}
      className="inner"
      role="progressbar"
      style={
        Object {
          "transform": "translateX(-67%)",
        }
      }
    />
  </div>
</ProgressBarBase>
`;

exports[`updates CSS transform based on \`width\` 2`] = `
<ProgressBarBase
  width={50}
>
  <div
    className="outer"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={50}
      className="inner"
      role="progressbar"
      style={
        Object {
          "transform": "translateX(-50%)",
        }
      }
    />
  </div>
</ProgressBarBase>
`;
